diff --git a/util/symcc_fuzzing_helper/src/main.rs b/util/symcc_fuzzing_helper/src/main.rs
index 6d284e1..04cd0e2 100644
--- a/util/symcc_fuzzing_helper/src/main.rs
+++ b/util/symcc_fuzzing_helper/src/main.rs
@@ -52,6 +52,10 @@ struct CLI {
 
     /// Program under test
     command: Vec<String>,
+
+    /// Number of seeds to run
+    #[structopt(short = "s")]
+    seed_count: usize,
 }
 
 /// Execution statistics.
@@ -301,7 +305,7 @@ fn main() -> Result<()> {
     let afl_config = AflConfig::load(options.output_dir.join(&options.fuzzer_name))?;
     log::debug!("AFL configuration: {:?}", &afl_config);
     let mut state = State::initialize(symcc_dir)?;
-
+    let mut count = 0;
     loop {
         match afl_config
             .best_new_testcase(&state.processed_files)
@@ -320,6 +324,10 @@ fn main() -> Result<()> {
             }
             state.last_stats_output = Instant::now();
         }
+        count += 1;
+        if count >= options.seed_count {
+            return Ok(());
+        }
     }
 }